File: api-cmssignature/api-mssclient/vodafonemssprovider/csharp/src/provider/VodafoneMSSProvider.cs
using tr.gov.tubitak.uekae.esya.api.common;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.provider;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.profile;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.signature;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.status;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.vodafone.profile;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.vodafone.signature;
using tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper;

namespace tr.gov.tubitak.uekae.esya.api.webservice.mssclient.vodafone.provider
{
    public class VodafoneMSSProvider: IMSSProvider
    {
        public IProfileRequest getProfileRequester(MSSParams aParams)
        {
            return new VodafoneProfileRequest(aParams);
        }

        public IStatusRequest getStatusRequester(MSSParams aParams)
        {
            throw new ESYAException("Not supported function!");
        }

        public ISignatureRequest getSignatureRequester(MSSParams aParams)
        {
            return new VodafoneSignatureRequest(aParams);
        }
    }
}

File: api-cmssignature/api-mssclient/vodafonemssprovider/src/tr/gov/tubitak/uekae/esya/api/webservice/mssclient/vodafone/provider/VodafoneMSSProvider.java
package tr.gov.tubitak.uekae.esya.api.webservice.mssclient.vodafone.provider;

import tr.gov.tubitak.uekae.esya.api.common.ESYAException;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.provider.IMSSProvider;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.profile.IProfileRequest;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.signature.ISignatureRequest;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.transaction.status.IStatusRequest;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.vodafone.profile.VodafoneProfileRequest;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.vodafone.signature.VodafoneSignatureRequest;
import tr.gov.tubitak.uekae.esya.api.webservice.mssclient.wrapper.MSSParams;

public class VodafoneMSSProvider implements IMSSProvider {

    @Override
    public IProfileRequest getProfileRequester(MSSParams aParams) {
        return new VodafoneProfileRequest(aParams);
    }

    @Override
    public IStatusRequest getStatusRequester(MSSParams aParams) throws ESYAException {
        throw new ESYAException("Not supported function!");
    }

    @Override
    public ISignatureRequest getSignatureRequester(MSSParams aParams) throws ESYAException {
        return new VodafoneSignatureRequest(aParams);
    }
}
